ipn/ipnlocal/local.go

```diff
@@ -3535,6 +3535,38 @@ func (b *LocalBackend) setTCPPortsInterceptedFromNetmapAndPrefsLocked() {
 	b.setTCPPortsIntercepted(handlePorts)
 }
 
+// SetServeConfig establishes or replaces the current serve config.
+//
+// b.mu must be held.
+func (b *LocalBackend) SetServeConfig(config *ipn.ServeConfig) error {
+	b.mu.Lock()
+	defer b.mu.Unlock()
+
+	nm := b.netMap
+	profileID := fmt.Sprintf("node-%s", nm.SelfNode.StableID) // TODO(maisem,bradfitz): something else?
+	confKey := ipn.ServeConfigKey(profileID)
+
+	j, err := json.Marshal(config)
```

Comment:
if config is nil, you want to WriteState nil, not []byte("null") probably.

(probably harmless, but a bit gross to accumulate "null" instead of clearing)